File: 2017 - 2/Programação de Computadores/All Netbeans Projects/Java/src/br/ufscar/dc/pc/Pintura.java
package br.ufscar.dc.pc;

/**
 * Classe Pintura
 *
 * @author Delano Medeiros Beder
 */
public class Pintura extends ObraDeArte {

    /*
     * Declaração dos atributos da classe
     */
    private String tipo;

    /*
     * Declaração do construtor da classe
     */
    public Pintura(String título, String artista, String material, int ano, 
            String tipo) {
        super(título, artista, material, ano);
        this.tipo = tipo;
    }

    /*
     * Declaração dos métodos da classe
     */
    @Override
    public void imprime() {
        System.out.println("=======================================");
        System.out.println("=             PINTURA                 =");
        System.out.println("=======================================");
        System.out.println("Título   : " + this.getTítulo());
        System.out.println("Artista  : " + this.getArtista());
        System.out.println("Material : " + this.getMaterial());
        System.out.println("Ano      : " + this.getAno());
        System.out.println("Tipo     : " + this.tipo);
    }
}

File: 2017 - 2/Programação de Computadores/All Netbeans Projects/C++/Pintura.cpp
#include "Pintura.h"

#include <iostream>

using namespace std;

namespace pc {

    Pintura::Pintura(string titulo, string artista, string material, int ano, string tipo) :
    ObraDeArte(titulo, artista, material, ano), tipo(tipo) {
    }

    Pintura::~Pintura() {
    }

    string Pintura::getTipo() const {
        return tipo;
    }

    void Pintura::imprime() {
        cout << "=======================================" << endl;
        cout << "=             PINTURA                 =" << endl;
        cout << "=======================================" << endl;
        cout << "Título   : " + this->getTitulo() << endl;
        cout << "Artista  : " + this->getArtista() << endl;
        cout << "Material : " + this->getMaterial() << endl;
        cout << "Ano      : ";
        cout << this->getAno() << endl;
        cout << "Tipo     : " + this->getTipo() << endl;
    }

}